lib/class-wp-rest-url-details-controller.php

```diff
@@ -77,7 +77,17 @@ public function parse_url_details( $request ) {
 		$data = $this->get_cache( $cache_key );
 
 		// Return cache if valid data.
-		if ( apply_filters( 'rest_url_details_use_cache', true ) && ! is_wp_error( $data ) && ! empty( $data ) ) {
+
+		/**
+		 * Filters whether to check for previously cached
+		 * data for a given URL. Filtering for `false` will
+		 * mean that the cache is bypassed.
+		 *
+		 * @param bool $use_cache whether or not to use the cache
+		 * @param string $url the attempted URL.
+		 * @param string $cache_key the generated cache key for the given URL.
+		 */
+		if ( apply_filters( 'rest_url_details_use_cache', $use_cache = true, $url, $cache_key ) && ! is_wp_error( $data ) && ! empty( $data ) ) {
```

Comment:
```suggestion
		if ( apply_filters( 'rest_url_details_use_cache', true, $url, $cache_key ) && ! is_wp_error( $data ) && ! empty( $data ) ) {
```

what's the use case for this filter though? Also, still would not check for `WP_Error` instances, as it leads to false assumptions. If anything I'd use `is_array()`

Reply:
> what's the use case for this filter though?

If I'm completely honest I added the filter as a way to aid with testing. I acknowledge that isn't a good reason to alter the system under test but it does make things a lot simpler. Also I feel there will be scenarios (yet to be identified) where folks don't want to use the cache under certain constraints. With the filter, they can check for specific `$url`s and opt in/out of cache as required.

Thinking about this again however, I could simply use [the existing filters in `get_transient()`](https://core.trac.wordpress.org/browser/tags/5.6/src/wp-includes/option.php#L825) within the tests to force return an empty value. That would remove the need for this `rest_url_details_use_cache` filter if we feel it is unnecessary. 

> Also, still would not check for WP_Error instances, as it leads to false assumptions. If anything I'd use is_array()

The argument about leading to false assumptions is a good one. I will update to `is_array()` as you suggest here.

Reply:
I've removed the filter and utilised the built in mechanism in transients to return a known value to avoid using the cache during the tests.

I have however, added a new tests to tests that if the cache value exists the the code will actually use it. Again we use the same mechanic to do this.

Let me know what you think and whether you'd be happy to 👍 this now.